src/libsync/creds/oauth.cpp

```diff
@@ -286,27 +294,20 @@ void OAuth::startAuthentication()
                         finalize(socket, accessToken, refreshToken, user, messageUrl);
                         return;
                     }
-                    // If the reply don't contains the user id, we must do another call to query it
-                    QNetworkRequest req;
-                    // We are not connected yet so we need to handle the authentication manually
-                    req.setRawHeader("Authorization", "Bearer " + accessToken.toUtf8());
-                    // We just added the Authorization header, don't let HttpCredentialsAccessManager tamper with it
-                    req.setAttribute(HttpCredentials::DontAddCredentialsAttribute, true);
-                    JsonApiJob *job = new JsonApiJob(_account->sharedFromThis(), QStringLiteral("ocs/v2.php/cloud/user"), {}, req, this);
-                    job->setTimeout(qMin(job->timeoutSec(), 30s));
-                    connect(job, &JsonApiJob::finishedSignal, this, [=] {
-                        if (job->ocsStatus() != 200) {
+
+                    auto *job = DetermineUserJobFactory(_networkAccessManager, accessToken, this).startJob(_serverUrl);
+
+                    connect(job, &Job::finished, this, [=]() {
+                        if (!job->success()) {
                             httpReplyAndClose(socket, QByteArrayLiteral("500 Internal Server Error"),
-                                tr("<h1>Login Error</h1><p>Failed to retrieve user info</p>").toUtf8());
+                                tr("<h1>Login Error</h1><p>%1</p>").arg(job->errorMessage()).toUtf8());
                             emit result(Error);
                         } else {
-                            const QString user = job->data().value(QStringLiteral("ocs")).toObject().value(QStringLiteral("data")).toObject().value(QStringLiteral("id")).toString();
+                            const auto user = qvariant_cast<QString>(job->result());
```

Comment:
result toString? I'm not sure which one feels better...

And in general I try to not use auto with QStrings as concatenation with the string builder and auto can go 💥 .
If we always stick to QString that can't happen by accident.